etl/regenstrief_s3_sftp.py

```diff
@@ -78,68 +67,107 @@ def lambda_handler(event, context):
 
     with transport:
         for s3_file in s3_files(event):
-            filename = s3_file.key
-            bucket = s3_file.bucket_name
-            contents = ""
+
+            # Get the current date in YYYYMMDD format
+            current_date = strftime("%Y_%m_%d")
+            filename = f"midrc_input/{current_date}/{s3_file.key}"
             try:
-                logger.info(f"S3-SFTP: Transferring S3 file '{s3_file.key}'")
+                logger.info(f"S3-SFTP: Transferring S3 file '{filename}' to SFTP")
                 transfer_file(sftp_client, s3_file, filename)
-            except botocore.exceptions.BotoCoreError as ex:
+            except BotoCoreError as ex:
                 logger.exception(
-                    f"S3-SFTP: Error transferring S3 file '{s3_file.key}'."
+                    f"S3-SFTP: Error transferring S3 file '{filename}'.\nException: {ex}"
                 )
-                contents = str(ex)
-                filename = filename + ".x"
+
+
+def connect_to_sftp(hostname, port, username, password, pkey):
+    """Connect to SFTP server and return client object."""
+    transport = paramiko.Transport((hostname, port))
+
+    k = paramiko.RSAKey.from_private_key_file(pkey) if pkey else None
+    transport.connect(username=username, password=password, pkey=k)
+    client = paramiko.SFTPClient.from_transport(transport)
+
+    logger.debug(f"S3-SFTP: Connected to remote SFTP server")
+    return client, transport
 
 
 def s3_files(event):
     """
-    Iterate through event and yield boto3.Object for each S3 file created.
+    Yields a boto3 S3 Object for each file created in an S3-triggered Lambda event.
+
     This function loops through all the records in the payload,
     checks that the event is a file creation, and if so, yields a
     boto3.Object that represents the file.
-    NB Redshift will trigger an `ObjectCreated:CompleteMultipartUpload` event
-    will UNLOADing the data; if you select to dump a manifest file as well,
-    then this will trigger `ObjectCreated:Put`
+
+    Note:
+    * Redshift will trigger an `ObjectCreated:CompleteMultipartUpload` event
```

Comment:
We don't even use Redshift in this process, right? 🤔 

in any case the Redshift-specific comment about "if you select to dump a manifest file as well" might be relevant and maybe should stay

Reply:
I don’t believe we need Redshift at all, and I wasn’t sure why that comment was added in the first place. I wasn’t certain if I should remove it entirely, so I just cleaned it up a bit instead.